### PR TITLE
Implement FrameBufferScale for IBufferedDrawable/BufferedContainer

### PR DESCRIPTION
**Pull request by smoogipoo**
Prereqs:
- [x] https://github.com/ppy/osu-framework/pull/3032

Demo: https://streamable.com/oo2g2

osu! side, this will allow drawing backgrounds at a lower scale to save on performance.

### CODE REVIEW

File: osu.Framework/Graphics/BufferedDrawNode.cs [hunk 1]
```diff
@@ -52,8 +53,9 @@ public override void ApplyState()
             backgroundColour = Source.BackgroundColour;
             screenSpaceDrawRectangle = Source.ScreenSpaceDrawQuad.AABBFloat;
             DrawColourInfo = Source.FrameBufferDrawColour ?? new DrawColourInfo(Color4.White, base.DrawColourInfo.Blending);
+            frameBufferScale = Source.FrameBufferScale;
 
-            frameBufferSize = new Vector2(MathF.Ceiling(screenSpaceDrawRectangle.Width), MathF.Ceiling(screenSpaceDrawRectangle.Height));
+            frameBufferSize = new Vector2(MathF.Ceiling(screenSpaceDrawRectangle.Width * frameBufferScale.X), MathF.Ceiling(screenSpaceDrawRectangle.Height * frameBufferScale.X));
```
Comment (swoolcock) [inline osu.Framework/Graphics/BufferedDrawNode.cs]: Assuming we want to allow individual scaling in both directions, the second here should be `frameBufferScale.Y`.

Comment (smoogipoo) [reply osu.Framework/Graphics/BufferedDrawNode.cs]: Whoops, nice catch.

**Comment by peppy** [comment]
Interface defaults don't seem to be working.

**Comment by huoyaoyuan** [comment]
According to [sharplab](https://sharplab.io/#v2:C4LgTgrgdgPgAgJgIwFgBQcDMACR2DC2I2AkugN7rbW44CWUw2AGtgLwB82CA3FTf2qIEg7JTQ1JuJAE4AFHKgBTAO4E5ASmwBDAM6kNAOmYa+EmgF90VtOizYGwJWABm2gMZLSFUY5bsuJDMLIA), the most institutive syntax should be correct.

**Pull request merged**